File: frontend/src/app/controller/service/admin/commun/DepotAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {DepotDto} from 'src/app/controller/model/commun/Depot.model';
import {DepotCriteria} from 'src/app/controller/criteria/commun/DepotCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class DepotAdminService extends AbstractService<DepotDto, DepotCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/depot/');
    }

    public constrcutDto(): DepotDto {
        return new DepotDto();
    }

    public constrcutCriteria(): DepotCriteria {
        return new DepotCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/stocky/service/facade/admin/commun/DepotAdminService.java
package ma.zs.stocky.service.facade.admin.commun;

import java.util.List;
import ma.zs.stocky.bean.core.commun.Depot;
import ma.zs.stocky.dao.criteria.core.commun.DepotCriteria;
import ma.zs.stocky.zynerator.service.IService;

import ma.zs.stocky.ws.dto.commun.DepotDto;
import org.springframework.http.HttpEntity;


public interface DepotAdminService extends  IService<Depot,DepotCriteria>  {



    HttpEntity<byte[]> createPdf(DepotDto dto) throws Exception;

}
